src/components/SubNav/SubNav.stories.tsx

```diff
@@ -0,0 +1,569 @@
+import type { Meta, StoryObj } from "@storybook/react";
+import Heading from "../Heading/Heading";
+import Hero from "../Hero/Hero";
+import Icon from "../Icons/Icon";
+
+import SubNav, {
+  SubNavButton,
+  SubNavLink,
+  actionBackgroundColorsArray,
+  highlightColorsArray,
+} from "./SubNav";
+
+const meta: Meta<typeof SubNav> = {
+  title: "Components/Navigation/SubNav",
+  component: SubNav,
+  argTypes: {
+    className: { control: false },
+    actionBackgroundColor: {
+      control: "select",
+      options: actionBackgroundColorsArray,
+      defaultValue: { summary: "blogs" },
+    },
+    highlightColor: {
+      control: "select",
+      options: highlightColorsArray,
+      defaultValue: { summary: "blogs" },
+    },
+    id: { control: false },
+    secondaryActions: {},
+    primaryActions: {},
+  },
+};
+
+const subNavButtonOnClick = (msg: string = "the button was clicked") => {
+  console.log(`SubNav button click message: ${msg}`);
+};
+
+export const WithControls: Story = {
+  args: {
+    className: undefined,
+    actionBackgroundColor: undefined,
+    highlightColor: undefined,
+    id: "subnav-with-controls",
+  },
+
+  render: (args: any) => (
+    <SubNav
+      {...args}
+      primaryActions={
+        <>
+          <SubNavButton
+            buttonType="text"
```

Comment:
You should not pass `buttonType="text"` since this is already set internally in the `SubNavButton` component. So you can remove this prop from all examples.